wlauto/external/uiauto/src/com/arm/wlauto/uiauto/BaseUiAutomation.java

```diff
@@ -37,6 +38,28 @@
 
 public class BaseUiAutomation extends UiAutomatorTestCase {   
 
+    public class ActionLogger {
+
+        private String testTag;
+        private Bundle parameters;
+
+        public ActionLogger(String testTag, Bundle parameters) {
+            this.testTag = testTag;
+            this.parameters = parameters;
+        }
+
+        public void start() {
+            if (Boolean.parseBoolean(parameters.getString("dumpsys_enabled"))) {
```

Comment:
I would parse the parameters inside the constructor and save the parsed values as members, it will result in both clearer code, and early failure in case of bad parameter values been passed.